src/System.Windows.Forms.Primitives/src/Interop/User32/Interop.EnumChildWindows.cs

```
@@ -50,7 +50,12 @@ public static BOOL EnumChildWindows(HandleRef hwndParent, EnumChildWindowsCallba
 
         private static BOOL HandleEnumChildWindowsNativeCallback(IntPtr hWnd, IntPtr lParam)
         {
-            return ((EnumChildWindowsCallback)GCHandle.FromIntPtr(lParam).Target)(hWnd);
+            if (GCHandle.FromIntPtr(lParam).Target is EnumChildWindowsCallback callback)
+            {
+                return callback(hWnd);
+            }
+
+            return BOOL.FALSE;
```

@JeremyKuhne an alternative was to do this:
```cs
return ((EnumChildWindowsCallback)GCHandle.FromIntPtr(lParam).Target!)(hWnd);
```
...but didn't feel right to me.

Thoughts?

---

The bang in this case makes sense. Everything is in this file and we'll fall over hard at runtime if we mess it up inadvertently, which I'd prefer to silently eating the mistake.

---
In reply to: [440574615](https://github.com/dotnet/winforms/pull/3451#discussion_r440574615) [](ancestors = 440574615)